cvejob/selectors/basic.py

```diff
@@ -1,286 +1,80 @@
 """This module contains default package name selector."""
 
-import itertools
-import functools
 import logging
-import re
 
-from nvdlib import utils
-
-from cvejob.config import Config
-from cvejob.outputs.victims import get_victims_affected_notation, get_victims_fixedin_notation
 from cvejob.version import BenevolentVersion
-from cvejob.utils import sort_versions
-
-from f8a_utils.versions import (
-    get_versions_for_pypi_package,
-    get_versions_for_npm_package,
-    get_versions_for_maven_package
+from cvejob.version_utils import (
+    get_upstream_versions, get_configuration_nodes, get_version_ranges, VersionSpec
 )
 
 
 logger = logging.getLogger(__name__)
 
 
-# noinspection PyTypeChecker
 class VersionSelector(object):
     """Selectors which picks winners based on existence of versions mentioned in the CVE record."""
 
-    def __init__(self, cve_doc, candidates):
-        """Constructor."""
-        self._doc = cve_doc
-        self._cve = self._doc.cve
+    def __init__(self, cve, candidates, ecosystem):
+        """Instantiate VersionSelector().
+
+        :param cve: nvdlib.model.Document, Document object encapsulating CVE information
+        :param candidates: list[cvejob.cpe2pkg.PackageNameCandidate],
+            a list of package name candidates
+        :param ecosystem: str, ecosystem name
+        """
+        self.cve = cve
 
-        self._candidates = candidates
+        self.candidates = candidates
+        self.ecosystem = ecosystem
 
     def pick_winner(self):
         """Pick single winner.
 
         Or no winner, if all candidates fail the version check.
+        :return: cvejob.cpe2pkg.PackageNameCandidate, or None
         """
-        candidate = None
-        safe_version_range = list()
-        affected_version_range = list()
+        if not self.candidates:
+            return None
 
-        cpe_entries = self._get_cpe_entries()
+        cve_versions = self.get_versions_from_cve(self.cve)
+        if not cve_versions:
+            return True
```

Comment:
is this the expected return value and type?

Reply:
Good catch. If there are no versions in CVE (rare), then there is nothing we can evaluate in this selector. So we should just return the candidate with highest confidence score. Let me fix this :+1: